### DISCUSSION TITLE
28-maps-memory-leaks/

### DISCUSSION THREAD

**Discussion by giscus** [teivah/100-go-mistakes#71] [Discussions]
# 28-maps-memory-leaks/

None

https://100go.co/28-maps-memory-leaks/

**Comment by fseda** [comment]
Are there any drawbacks to storing pointers instead of the actual value?

**Reply by scooby** [reply]
There are the usual problems with using pointers: you have to guard against nils, accessing values requires another pointer dereference, and it puts more load on the GC.

**Comment by nurjamil** [comment]
does it still happening after go 1.21 ?

**Reply by scooby** [reply]
Yes, [maps still don't automatically release memory](https://github.com/golang/go/issues/20135).

**Reply by LordMoMA** [reply]
why 1.16 does not have this problem?

**Comment by LordMoMA** [comment]
i did not see this issue after 1.23

**Reply by LordMoMA** [reply]
tested with:

```
➜  go-learning gvm use go1.16
Now using version go1.16
➜  go-learning go run .
Alloc = 0 MiB
Alloc = 461 MiB
Alloc = 0 MiB
```

even 1.16 did not have this problem.

**Reply by gevorg-gevorgyan-idt** [reply]
@LordMoMA 
Probably you missed runtime.KeepAlive(m)  
And GC is cleaning whole map. 
I see same issue with 1.26 , little bit improved footprint 288MB instead of 461MB.
Alloc = 0 MB
Alloc = 288 MB
Alloc = 287 MB

**Comment by vincenshen01** [comment]
It's very useful for me to understand underlaying principles of golang.

**Comment by AmanKhys** [comment]
I went along with the code and you're wrong about it as of date.
The GC has fixed this and after deleting all the elements and then running a runtime.GC()
it clears everything.

One way to help me out, an internship offer if possible working with go.
amankhys2112@gmail.com
